sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -367,6 +368,38 @@ private[sql] class RocksDBStateStoreProvider
   private def verify(condition: => Boolean, msg: String): Unit = {
     if (!condition) { throw new IllegalStateException(msg) }
   }
+
+  override def replayStateFromSnapshot(startVersion: Long, endVersion: Long): StateStore = {
+    try {
+      if (startVersion < 1) {
+        throw QueryExecutionErrors.unexpectedStateStoreVersion(startVersion)
+      }
+      if (endVersion < startVersion) {
+        throw QueryExecutionErrors.unexpectedStateStoreVersion(endVersion)
+      }
+      rocksDB.loadFromSnapshot(startVersion, endVersion)
```

Comment:
just to double check, readOnly flag is not needed unlike the path of load(), do I understand correctly? If then shall we just implement one of two and call other to reduce redundant code?

Reply:
Yes. Good catch.